include/flatbuffers/flatbuffers.h

```diff
@@ -179,6 +180,25 @@ template<typename T, typename IT> struct VectorIterator {
   const uint8_t *data_;
 };
 
+template<typename Iterator> struct VectorReverseIterator :
+  public std::reverse_iterator<Iterator> {
+
+  explicit VectorReverseIterator(Iterator iter) : iter_(iter) {}
+
+  typename Iterator::value_type operator*() const {
+    Iterator tmp = iter_;
+    return *--tmp;
```

Comment:
I presume this is somehow better/faster than just `*(iter - 1)` ?

Reply:
I think you're right, I should be using `*(iter_ - 1)`. But I also need to add a `const` keyword on the `VectorIterator::operator-(const uoffset_t &)` function to make it work.

I'll make the changes asap.

Reply:
I've updated my PR. Thanks for the comment.